src/draw/lv_draw_buf.h

```diff
@@ -25,17 +25,127 @@ extern "C" {
 #define LV_STRIDE_AUTO 0
 LV_EXPORT_CONST_INT(LV_STRIDE_AUTO);
 
+#define _LV_ZOOM_INV_UPSCALE 5
+
+/** Magic number for lvgl image, 9 means lvgl version 9
+ *  It must not be a valid ASCII character nor larger than 0x80. See `lv_image_src_get_type`.
+ */
+#define LV_IMAGE_HEADER_MAGIC (0x19)
+LV_EXPORT_CONST_INT(LV_IMAGE_HEADER_MAGIC);
+
 /**********************
  *      TYPEDEFS
  **********************/
 
+typedef enum _lv_image_flags_t {
```

Comment:
Well, I think removing `lv_image_buf`, then simply copy-pasting it's enums and things is not the best way.
This change made code more confusing: why are there "image" related enums in `lv_draw_buf` ?
I think we should keep the original `lv_image_buf`, or move into `lv_image` OR change the "image" names here in `lv_draw_buf` to avoid this deceptive naming here.
But I think sometimes it's better not to touch an already working and clean code.... there is no need to refactor everything.

Reply:
See discussion here. https://github.com/lvgl/lvgl/issues/5054

The primary distinction to note is that lv_image_dsc_t is intended for image resources, while the draw buffer is used for internally allocated intermediate images. It's crucial to maintain this separation clear to prevent confusion about the appropriate usage and timing for each.

Removing the lv_image_buf.c/.h is because we can only do modification to draw buffer. 
The image flag  in draw buffer does confuse a little. Do you have better idea?

Reply:
I also find "image flags" in "draw buf" a little bit confusing. However I understand that it's in line wit our endeavor to use only `draw_buf` in the end. 

Can we have e.g. `LV_IMAGE_FLAGS_PREMULTIPLIED` or other flags on a draw buf?

If so, to proceed gradually I suggest:
- Renaming everything in this file to "... draw buf ..." 
- Add the image related names to `lv_api_map.h` or to this file with typedef an defines.

Reply:
Well, I think it's not ideal. Because a draw buffer is an image, so it can have image flags. 
An image is not an draw buffer, but it can have image flags since it's an image.

It has similar idea in OpenGL, see https://stackoverflow.com/questions/13909404/whats-the-difference-between-image-and-texture-in-opengl. An image loaded to RAM that GPU can use later is a texture. We use name draw buffer instead.

Reply:
my concern is not only about image flags, but also the other image structs moved to this draw buf file.
image header struct is also similar to draw buf struct, and some other functions are moved also to here, still with `lv_image_` prefix.
It makes confusing to differ the “real image” from a “simple draw buffer”.

btw the binary images (stored in C array or separate file) will also contain stride, and “empty vertical space” (unused bytes), when image width does not fit exactly in 4/8/16 bytes alignment?

Reply:
I actually wanted to rename it to lv_draw_buf_xxx, because only draw buffer can be modifiable, but am concerned about API breaking.  Do you think we should rename them too?

> btw the binary images (stored in C array or separate file) will also contain stride, and “empty vertical space” 
 (unused bytes), when image width does not fit exactly in 4/8/16 bytes alignment?

The padding is only added as instructed in the python script. It's also not needed by the software renderer and image decoder.

Reply:
The image related API now goes out of lv_draw_buf.h